.cargo/config.toml

```diff
@@ -0,0 +1,32 @@
+#
+# An auto defined `clippy` feature was introduced,
+# but it was found to clash with user defined features,
+# so was renamed to `cargo-clippy`.
+#
+# If you want standard clippy run:
+# RUSTFLAGS= cargo clippy
+[target.'cfg(feature = "cargo-clippy")']
+rustflags = [
+  "-Aclippy::all",
+  "-Dclippy::correctness",
+  "-Aclippy::if-same-then-else",
+  "-Aclippy::clone-double-ref",
+  "-Dclippy::complexity",
+  "-Aclippy::zero-prefixed-literal",     # 00_1000_000
+  "-Aclippy::type_complexity",           # raison d'etre
+  "-Aclippy::nonminimal-bool",           # maybe
+  "-Aclippy::borrowed-box",              # Reasonable to fix this one
+  "-Aclippy::too-many-arguments",        # (Turning this on would lead to)
+  "-Aclippy::unnecessary_cast",          # Types may change
+  "-Aclippy::identity-op",               # One case where we do 0 +
+  "-Aclippy::useless_conversion",        # Types may change
```

Comment:
Looks like these lints were copied from Substrate. But it makes me sad that this is `allow`. We have a bunch of easy simplifications like e.g.

```
  --> xcm/src/v1/junction.rs:95:66
   |
95 |             Junction0::Plurality { id, part } => Ok(Self::Plurality { id: id.into(), part }),
   |                                                                           ^^^^^^^^^ help: consider removing `.into()`: `id`
```

This lint can be annoying during development as "Types may change", but in CI it should be more strict IMO. That would keep friction to a minimum.

Probably this should be a follow-up issue where people can chime in. Super low priority in the grand scheme of things, so let's leave the lints as-is for now.

Reply:
That also really depends on people's workflow. I only run clippy manually when I'm done. I.e. that lint wouldn't bother me at all.

Reply:
Yeah, I guess you can just have clippy in your pre-push hook instead of pre-commit.

Also I remembered you can `#[allow(clippy::lint)]` on certain lines. I would prefer that as opposed to white-listing a lint for the whole codebase for "One case where we do 0 +".